lib/LANraragi/Model/ReadingGroup.pm

```diff
@@ -0,0 +1,229 @@
+package LANraragi::Model::ReadingGroup;
+
+use strict;
+use warnings;
+use utf8;
+
+use Redis;
+use Mojo::JSON qw(decode_json encode_json);
+
+use LANraragi::Utils::Database qw(redis_encode redis_decode invalidate_cache get_archive_json_multi);
+use LANraragi::Utils::Logging qw(get_logger);
+
+# get_reading_group_list()
+#   Returns a list of all the reading group objects.
+sub get_reading_group_list {
+
+    my $redis = LANraragi::Model::Config->get_redis;
+
+    # Categories are represented by RG_[timestamp] in DB. Can't wait for 2038!
+    my @rgs = $redis->keys('RG_??????????');
+
+    # Jam categories into an array of hashes
+    my @result;
+    foreach my $key (@rgs) {
+        my %data = get_reading_group($key);
+        push( @result, \%data );
+    }
+
+    # # Only get the first X keys
+    # my $keysperpage = LANraragi::Model::Config->get_pagesize;
+
+    # # Return total keys and the filtered ones
+    # my $end = min( $start + $keysperpage - 1, $#filtered );
+    # return ( $total, $#filtered + 1, @filtered[ $start .. $end ] );
+
+    return @result;
+}
+
+# create_reading_group(name, existing_id)
+#   Create a Reading Group.
+#   If an existing Reading Group ID is supplied, said Reading Group will be updated with the given parameters.
+#   Returns the ID of the created/updated Reading Group.
+sub create_reading_group {
+
+    my ( $name, $rg_id ) = @_;
```

Comment:
If you point to a module that has that new style, I can refactor all my code to ensure it works similarly.

Reply:
Look at https://github.com/Difegue/LANraragi/blob/dev/lib/LANraragi/Utils/Archive.pm for an example.  
You'll have to declare 
```
use feature qw(signatures);
no warnings 'experimental::signatures';
``` 
in the opening of any file with sigs otherwise old Perls will yell at you